idseq_pipeline/commands/accessionid2seq_functions.py

```diff
@@ -353,55 +370,57 @@ def get_sequence_by_accession_id_ntf(accession_id, nt_loc_dict, ntf):
         (seq_name, ref_seq) = ntf.read(seq_len).split("\n", 1)
         ref_seq = ref_seq.replace("\n", "")
         seq_name = seq_name.split(" ", 1)[1]
-    return (ref_seq, seq_name)
+    return ref_seq, seq_name
 
 
 def get_sequence_by_accession_id_s3(accession_id, nt_loc_dict, nt_bucket,
                                     nt_key):
     seq_len = 0
     seq_name = ''
     entry = nt_loc_dict.get(accession_id)
-    if entry:
-        (range_start, name_length, seq_len) = entry
+    if not entry:
+        return seq_len, seq_name
+
+    (range_start, name_length, seq_len) = entry
 
-        accession_file = 'accession-%s' % accession_id
-        NUM_RETRIES = 3
-        for attempt in range(NUM_RETRIES):
+    accession_file = 'accession-%s' % accession_id
+    NUM_RETRIES = 3
+    for attempt in range(NUM_RETRIES):
+        try:
+            pipe_file = 'pipe-{attempt}-accession-{accession_id}'.format(
+                attempt=attempt, accession_id=accession_id)
+            os.mkfifo(pipe_file)
+            get_range = "aws s3api get-object --range bytes=%d-%d --bucket %s --key %s %s" % (
+                range_start, range_start + name_length + seq_len - 1,
+                nt_bucket, nt_key, pipe_file)
+            get_range_proc = subprocess.Popen(
+                get_range, shell=True, stdout=DEVNULL)
+
+            cmd = "cat {pipe_file} | tee >(tail -n+2 | tr -d '\n' > {accession_file}) | head -1".format(
+                pipe_file=pipe_file, accession_file=accession_file)
+            seq_name = subprocess.check_output(
+                cmd, executable='/bin/bash', shell=True).split(" ", 1)[1]
+            exitcode = get_range_proc.wait()
+            msg = "Error in get_sequence_by_accession_id_s3."
+            assert exitcode == 0, msg
+            seq_len = os.stat(accession_file).st_size
+            break
+        except:
+            if attempt + 1 < NUM_RETRIES:
+                time.sleep(1.0 * (4**attempt))
```

Comment:
can this benefit from the retry decorator?   i am not sure, just asking

Reply:
hm yeah that might be good